src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala

```diff
@@ -462,10 +463,21 @@ class HttpGoogleBillingDAO(appName: String,
           case ge: GoogleJsonResponseException if ge.getStatusCode == 400 && ge.getMessage.contains("while it has other pending delete operations") =>
         }
       }
-    } yield {
-      //nah
-    }
-  }
+    } yield {}
+
+  // Leonardo currently only created GCE VMs in zone us-central1-a. If it were to start supporting other zones, this will need to be updated.
+  def cleanupVMs(projectName: String): Future[Unit] =
+    for {
+      result <- googleRq(computeManager.instances().list(projectName, "us-central1-a"))
+      instances = googNull(result.getItems)
+      instanceNames = instances.map(i => i.getName)
+      _ <- sequentially(instanceNames) { instanceName =>
+        googleRq(computeManager.instances().delete(projectName, "us-central1-a", instanceName)).recover {
+          // ignore errors on already deleting clusters
+          case ge: GoogleJsonResponseException if ge.getStatusCode == 400 && ge.getMessage.contains("while it has other pending delete operations") =>
```

Comment:
I suspect this is a Dataproc-specific message, could you try to delete a VM twice, see what the error is the second time (if any), and recover on that?

Reply:
I tried this a couple times but it seems like the second call just stalls until the first is completed and then throws a `resource not found` error. Wasn't sure if I was missing something, and don't know if it's a good idea to recover on an error that's not specific to multiple delete requests

Reply:
Yeah I think we should probably recover on that then, the point is to recover here if a vm is already deleted (or in the process of being deleted), to ensure a failure doesn’t abort cleaning up other VMs (since it’s all done sequentially).

Also note that the gcloud CLI might have different behavior than the java client.

Reply:
I tested using the `cleanVMs` branch and it looks like there's no failure at all through the java client for two consecutive delete calls for the same instance. Removed the `recover`, can revisit if we see any different behavior
<img width="1676" alt="Screen Shot 2020-06-01 at 2 29 25 PM" src="https://user-images.githubusercontent.com/39962314/83441325-588a9600-a414-11ea-83c2-c18ee7006fba.png">

Reply:
thanks for checking!